server/service/src/sync/translations/invoice_line.rs

```diff
@@ -153,14 +153,25 @@ impl SyncTranslation for InvoiceLineTranslation {
                 (item.code, None, total, total)
             }
         };
-        // When invoice lines are coming from another site, we don't get stock line and location
-        // so foreign key constraint is violated, thus we want to set them to None if it's foreign site record.
-        let (stock_line_id, location_id) = if is_active_record_on_site(
+
+        let is_record_active_on_site = is_active_record_on_site(
             &connection,
             ActiveRecordCheck::InvoiceLine {
                 invoice_id: invoice_id.clone(),
             },
-        )? {
+        )?;
+
+        let is_stock_line_valid = match stock_line_id {
```

Comment:
Hmm, i don't actually agree with this change, I'll point it out in the issue you've created for mSupply (to get this PR merged), I think not having stock line referenced by active record on site should be logged and not silently ignored (previously row would have been missing but logged against sync_buffer)